addons/mail/models/mail_mail.py

```diff
@@ -100,6 +100,11 @@ def default_get(self, fields):
             self = self.with_context(dict(self._context, default_type=None))
         return super(MailMail, self).default_get(fields)
 
+    def mass_retry(self):
+        for mail in self:
+            if mail.state in ['exception', 'cancel']:
+                mail.mark_outgoing()
```

Comment:
as mark_outgoing is not an 'ensure_one' thing, you can directly do :
`self.filtered(lambda mail: mail.state in ['exception', 'cancel']).mark_outgoing()`